presentation-compiler/test/dotty/tools/pc/utils/JRE.scala

```diff
@@ -0,0 +1,22 @@
+package dotty.tools.pc.utils
+
+object JRE:
+
+  def getJavaMajorVersion: Int =
+    val javaVersion = sys.env.get("java.version").filter(!_.isEmpty())
+
+    javaVersion match
+      case Some(version) if version.startsWith("1.8") => 8
+      case _ =>
+        scala.util.Try:
+          val versionMethod = classOf[Runtime].getMethod("version")
+          versionMethod.nn.setAccessible(true)
+          val version = versionMethod.nn.invoke(null)
+
+          val majorMethod = version.getClass().getMethod("feature")
+          majorMethod.nn.setAccessible(true)
+          val major = majorMethod.nn.invoke(version).asInstanceOf[Int]
+          major
+        .getOrElse(8) // Minimal version supported by Scala
+
```

Comment:
It will if you manually reimplement the version parser, which also has 2 variants after  Java 10. This method will always get you the right JRE versions without risk of making the mistake and will not require additional implementation.

Reply:
Also, this is how popular testing libraries implement this check. We're sadly on JUnit 4, and it does not allow us to filter tests based on current Java version, as it was added in JUnit 5.

Reply:
Oh this is `java.specification.version`